docs/advanced/testing.md

```diff
@@ -0,0 +1,63 @@
+# Testing
+
+This document outlines the process of testing Polly's resilience pipelines. Instead of testing the internal behavior of resilience pipelines, focus on testing your configurations or custom delegates.
+
+To support this, Polly offers the [`Polly.Testing`](https://www.nuget.org/packages/Polly.Testing/) package, making the testing process more straightforward. This package provides a range of APIs designed to help you confirm the setup and combination of resilience pipelines in your user code.
+
+## Usage
+
+Begin by adding the `Polly.Testing` package to your test project:
+
+```sh
+dotnet add package Polly.Testing
+```
+
+Use the `GetPipelineDescriptor` extension method to get the `ResiliencePipelineDescriptor` which provides details on the pipeline's composition:
+
+<!-- snippet: get-pipeline-descriptor -->
+```cs
+// Build your resilience pipeline.
+ResiliencePipeline pipeline = new ResiliencePipelineBuilder()
+    .AddRetry(new RetryStrategyOptions
+    {
+        MaxRetryAttempts = 4
+    })
+    .AddTimeout(TimeSpan.FromSeconds(1))
+    .Build();
+
+// Retrieve the descriptor.
+ResiliencePipelineDescriptor descriptor = pipeline.GetPipelineDescriptor();
+
+// Check the pipeline's composition with the descriptor.
+Assert.Equal(2, descriptor.Strategies.Count);
```

Comment:
I would maybe add a check here that the first strategy is `RetryStrategy`? It is true that there is options check, but to me it fit the description.

Reply:
Not really possible because strategy implementations are internal. Checking for options type fulfils that role.

Reply:
ok I thought that might be the case, because I remember some talks about it. Maybe just tweak the "composition" word in the comment? "composition" implied to me that we are checking the structure.

Reply:
But we are checking the composition. For example:

``` csharp
// Verify the retry settings.
var retryOptions = Assert.IsType<RetryStrategyOptions>(descriptor.Strategies[0].Options);
Assert.Equal(4, retryOptions.MaxRetryAttempts);
```

This checks that the first strategy is retry and it has correct configuration.

Reply:
ok I see, so basically it was like title header for everything past it. That makes sense.